File: pinyougou_manager_web/src/main/webapp/js/conrtoller/brandcontroller.js
// 控制层
app.controller("brandController",function ($scope,$controller,brandService) {

    // 继承baseController
    $controller('baseController',{$scope:$scope});

    $scope.findAll = function () {
        brandService.findAll().success(
            function (response) {
                $scope.list = response;
            }
        )
    }


    //分页
    $scope.findPage=function(page,rows){
        brandService.findPage(page,rows).success(
            function(response){
                $scope.list=response.rows;
                $scope.paginationConf.totalItems=response.total;//更新总记录数
            }
        );
    }
    // 添加品牌
    $scope.save=function () {
        // 新增和修改公用save方法，定义变量进行区分
        /*var methodName = "add";*/
        var object = null;
        if($scope.entity.id != null){
            // 执行更新操作
            object = brandService.update($scope.entity);
        }
        else
        {
            object = brandService.add($scope.entity);
        }
        object.success(
            function (response) {
                if(response.flag){// 添加成功
                    $scope.reloadList();
                }
                else
                {// 添加失败
                    alert(response.message)
                }
            }
        )
    }
    //根据id查询数据
    $scope.findOne=function (id) {
        brandService.findOne(id).success(
            function (response) {
                $scope.entity=response;
            }
        )
    }

    // 条件查询
    $scope.searchEntity={};
    $scope.search=function (page,rows) {
        brandService.search(page,rows,$scope.searchEntity).success(
            function(response){
                $scope.list=response.rows;
                $scope.paginationConf.totalItems=response.total;//更新总记录数
            }
        );
    }
})

File: pinyougou_manager_web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.entity.PageResult;
import com.pinyougou.entity.Result;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    /**
     * 查询所有
     * @return
     */
    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){

        return brandService.findAll();
    }

    /**
     * 商品分页
     * @param page 当前页
     * @param rows 当前页显示的条数
     * @return
     */
    @RequestMapping("/findPage")
    public PageResult findPage(int page,int rows){

        return brandService.findPage(page,rows);
    }

    /**
     * 添加品牌
     * @param tbBrand
     * @return
     */
    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand tbBrand){

        return brandService.add(tbBrand);
    }

    /**
     * 根据主键查询单个实体
     * @param id
     * @return
     */
    @RequestMapping("/findOne")
   public TbBrand findOne(Long id){
        return brandService.findOne(id);
   }

    /**
     * 根据主键更新品牌
     * @param tbBrand
     * @return
     */
    @RequestMapping("/update")
   public Result update(@RequestBody TbBrand tbBrand){
       try {
           brandService.update(tbBrand);
           return new Result(true,"更新成功");
       } catch (Exception e) {
           e.printStackTrace();
           return new Result(false,"更新失败");
       }
   }

    /**
     * 删除选中项目
     * @param ids
     * @return
     */
    @RequestMapping("/delete")
   public Result delete(Long[] ids){
        try {
            brandService.delete(ids);
            return new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(true,"删除失败");
        }
    }
   @RequestMapping("/search")
   public PageResult search(@RequestBody TbBrand tbBrand,int page,int rows){

         return brandService.findPage(tbBrand,page,rows);
   }

    /**
     * 查询品牌项
     * @return
     */
    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList(){

        return brandService.selectOptionList();
    }

}
